src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs

```diff
@@ -635,26 +744,29 @@ private void WriteProperties(IEnumerable properties)
                 return;
             }
 
+            if (nameValueListBuffer.Count > 0)
```

Comment:
super-nit: Just `Clear()` unconditionally for brevity?

Reply:
Clear() also increments the internal `version` field of the list. I'm sure it doesn't matter at all and it's OK either way, but I thought of the version field overflowing (it's an int32) and decided to do this instead :) Call me paranoid :)

Reply:
Haah, you're right. Simpler is better. Fixed.